src/gui/scene/scene.hpp

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include "../../world.hpp"
+#include "../data_structures/frame_buffer_multisample.hpp"
+#include "../data_structures/static_mesh.hpp"
+#include "../data_structures/terrain_mesh.hpp"
+#include "../render/quad_renderer_multisample.hpp"
+#include "../render/renderer.hpp"
+#include "../render/shadow_map.hpp"
+#include "../render/sky.hpp"
+
+#include <GLFW/glfw3.h>
+
+namespace gui {
+
+class Scene {
+ private:
+    World& world_;
+    data_structures::FrameBufferMultisample fbo;
+
+    render::MainRenderer MR;
+    render::ShadowMap SM;
+    render::QuadRendererMultisample QRMS;
+
+    render::SkyRenderer SR;
+
+    // TODO these should be saved in world
+    data_structures::StaticMesh treesMesh;
+
+ public:
+    Scene(
+        World& world_, uint32_t width, uint32_t height, uint32_t shadow_map_width_height
+    );
+    GLuint get_scene();
+    GLuint get_depth_texture();
+    uint32_t get_shadow_width();
+    uint32_t get_shadow_height();
+    void update(GLFWwindow* window);
+
+    std::vector<glm::ivec3> get_model_matrices_temp(World& world);
```

Comment:
Why is this `temp`?

Reply:
Because this is still a test tile. These get the transforms of the trees. The trees are not implimented correctly so this cursed function is used. All this should be fixed in Graphics/refactoring.

Reply:
Alright I'm trusting you.